File: infrastructures/ddd/python/src/api/dto/infra/base_request_dto.py
"""
Base class for all request DTOs.
Request DTOs are used for HTTP transport from client to server.
"""

from dataclasses import dataclass
from typing import Optional
import uuid


@dataclass
class BaseRequestDto:
    """
    Base class for all request DTOs.
    
    Request DTOs:
    - Live in /api/dto/requests/
    - Have Request suffix
    - Handle HTTP transport only
    - NO business logic
    - Edge validation happens here
    """
    
    request_id: str = ""
    
    def __post_init__(self):
        """Initialize request ID if not provided."""
        if not self.request_id:
            self.request_id = str(uuid.uuid4())

File: infrastructures/ddd/java/src/main/java/com/oddly/ddd/api/dto/v1/infra/BaseRequestDto.java
package com.oddly.ddd.api.dto.v1.infra;

/**
 * Base class for all request DTOs.
 * Request DTOs:
 * - Live in /api/dto/v1/requests/
 * - Have Request suffix
 * - Handle HTTP transport only
 * - NO business logic
 * - Include validation annotations
 */
public abstract class BaseRequestDto {
    private String m_requestId;

    public String getRequestId() {
        return m_requestId;
    }

    public void setRequestId(String p_requestId) {
        this.m_requestId = p_requestId;
    }
}
